### PR TITLE
update schema with new tables, insert new records in data.sql, and ru…

### PR DESCRIPTION
**Pull request by sulaimonaa**
Write queries to answer the following:
- Who was the last animal seen by William Tatcher?
- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- How many visits were with a vet that did not specialize in that animal's species?
- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

visit-num-maisy
visit-with-specialization
detail-most-recent-visit-animal
animal-visit-by-Maisy
animal-most-visited
animal-visit-by-stephanie-
list-vets-specialties
animals-seen-Stephanie
last-animal-seen-Williams

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -27,3 +27,27 @@ ALTER TABLE animals ADD COLUMN owners_id INT;
...
+    vet_id INT,
+    specialization VARCHAR(255),
+    FOREIGN KEY (vet_id) REFERENCES vets(id)
+);
+
+-- Create the visits table
+CREATE TABLE visits (
+    animal VARCHAR(255),
+    vet_id INT,
+    visit_date DATE,
```
Comment (bienvenuushindi) [inline schema.sql]: :red_circle: There is a many-to-many relationship between the tables animals and vets.  This "join table"  visits should contain two relationships [foreign keys] one for `vets` and another one for `visits`. Currrently  It has only one the vet_id. Kindly make sure you update animal column to reflect this requested changes. The animal column should be renamed to animal_id to make it consistent with the naming convention for primary keys.
```

CREATE TABLE visits (
    animal_id INT NOT NULL,
    vet_id INT NOT NULL,
    visit_date DATE,
    FOREIGN KEY (animal_id) REFERENCES animals(id),
    FOREIGN KEY (vet_id) REFERENCES vets(id)
);
```

File: schema.sql [hunk 2]
```diff
@@ -27,3 +27,27 @@ ALTER TABLE animals ADD COLUMN owners_id INT;
...
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255),
+    age INT,
+    date_of_graduation DATE
+);
+
+-- Create the specializations table
+CREATE TABLE specializations (
+    vet_id INT,
+    specialization VARCHAR(255),
```
Comment (bienvenuushindi) [inline schema.sql]: :red_circle: There is a many-to-many relationship between the tables species and vets.  This "join table"  specializations should contain two relationships one for `vets` and another one for `species`. Currently, It has only one vet_id. The specialization column should be renamed to species_id to make it consistent with the naming convention for primary keys.
```
 CREATE TABLE specializations (
   species_id int,
   vet_id int
 );

```

**Review by bienvenuushindi — CHANGES_REQUESTED** [review]
Hi @sulaimonaa  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project looks good:heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql [hunk 3]
```diff
@@ -27,3 +27,28 @@ ALTER TABLE animals ADD COLUMN owners_id INT;
...
+-- Create the specializations table
+CREATE TABLE specializations (
+    vet_id INT,
+    species_id INT,
+    FOREIGN KEY (vet_id) REFERENCES vets(id)
+);
+
+-- Create the visits table
+CREATE TABLE visits (
+    animal VARCHAR(255),
```
Comment (jerryowusu) [inline schema.sql]: As requested by the previous reviewer, you do not need the animal varchar column here, but rather the animal_id which should be an `int` just like the vet_id. Kindly consider removing this line and updating it with the animal id (`animal_id int`)

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @sulaimonaa 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

noooo


## Highlights
Neat code :+1: 
Great effort so far :+1: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

- [ ] According to the project requirement you are suppose to use gitflow, but you have made the pull request from a feature branch to the main branch which is not gitflow. Kindly ensure you are using the correct git flow method. 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @sulaimonaa , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested change. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**